src/averageAges.js

```diff
@@ -20,6 +20,29 @@ function calculateMenAverageAge(people, century) {
   // avoid using loop and forEach
   // replace `if ()` statement with &&, || or ?:
   // without nesting
+  const totalAge = people.map(human => human.sex === 'm'
+    ? human.died - human.born : 0);
+
+  const totalAgeCentury = people.map(
+    human => Math.ceil(human.died / 100) === century
+    && human.sex === 'm' ? human.died - human.born : 0
+  );
+
+  const womenFiltered = totalAge.filter(year => year > 0);
+
+  const filteredCentury = totalAgeCentury.filter(year => year > 0);
+
+  const avarageMaleAge = womenFiltered.reduce((a, b) => a + b, 0);
+
+  const avarageMaleAgeCentury = filteredCentury.reduce((a, b) => a + b, 0);
+
+  let final = 0;
+
+  century > 0 ? final
+    = avarageMaleAgeCentury / filteredCentury.length
```

Comment:
You are doing almost the same in each function (reduce and result of reduce / length). please, move it to a separate function

Reply:
done